python/ray/tune/BUILD

```diff
@@ -142,74 +178,348 @@ py_test(
     tags = ["exclusive"],
 )
 
+# --------------------------------------------------------------------
+# Tests from the python/ray/tune/tests directory.
+# Covers all remaining tests that do not start with `test_`.
+# Please keep these sorted alphabetically.
+# --------------------------------------------------------------------
 py_test(
-    name = "test_trainable_util",
+    name = "example",
     size = "small",
-    srcs = ["tests/test_trainable_util.py"],
+    srcs = ["tests/example.py"],
     deps = [":tune_lib"],
+    tags = ["exclusive", "example"],
 )
 
+# Todo: Ensure MPLBACKEND=Agg
 py_test(
-    name = "test_trial_scheduler",
+    name = "tutorial",
     size = "medium",
-    srcs = ["tests/test_trial_scheduler.py"],
+    srcs = ["tests/tutorial.py"],
     deps = [":tune_lib"],
-    tags = ["exclusive"],
+    tags = ["exclusive", "example"],
 )
 
+# --------------------------------------------------------------------
+# Examples from the python/ray/tune/examples directory.
+# Please keep these sorted alphabetically.
+# --------------------------------------------------------------------
 py_test(
-    name = "test_trial_scheduler_pbt",
+    name = "async_hyperband_example",
     size = "medium",
-    srcs = ["tests/test_trial_scheduler_pbt.py"],
+    srcs = ["examples/async_hyperband_example.py"],
     deps = [":tune_lib"],
-    tags = ["exclusive"],
+    tags = ["exclusive", "example"],
+    args = ["--smoke-test"]
 )
 
 py_test(
-    name = "test_tune_restore",
-    size = "large",
-    srcs = ["tests/test_tune_restore.py"],
+    name = "ax_example",
+    size = "medium",
+    srcs = ["examples/ax_example.py"],
     deps = [":tune_lib"],
-    tags = ["jenkins_only", "exclusive"],
+    tags = ["exclusive", "example", "py37"],
+    args = ["--smoke-test"]
 )
 
 py_test(
-    name = "test_tune_save_restore",
+    name = "bayesopt_example",
+    size = "medium",
+    srcs = ["examples/bayesopt_example.py"],
+    deps = [":tune_lib"],
+    tags = ["exclusive", "example"],
+    args = ["--smoke-test"]
+)
+
+py_test(
+    name = "bohb_example",
+    size = "medium",
+    srcs = ["examples/bohb_example.py"],
+    deps = [":tune_lib"],
+    tags = ["exclusive", "example"]
+)
+
+py_test(
+    name = "cifar10_pytorch",
+    size = "medium",
+    srcs = ["examples/cifar10_pytorch.py"],
+    deps = [":tune_lib"],
+    tags = ["exclusive", "example", "pytorch"],
+    args = ["--smoke-test"]
+)
+
+py_test(
+    name = "dragonfly_example",
+    size = "medium",
+    srcs = ["examples/dragonfly_example.py"],
+    deps = [":tune_lib"],
+    tags = ["exclusive", "example"],
+    args = ["--smoke-test"]
+)
+
+# Does not work without awscli tool installed. Check if can be mocked.
+# py_test(
+#     name = "durable_trainable_example",
+#     size = "medium",
+#     srcs = ["examples/durable_trainable_example.py"],
+#     deps = [":tune_lib"],
+#     tags = ["exclusive", "example"],
+#     args = ["--local", "--mock-storage"]
+# )
+
```

Comment:
@richardliaw can you look at this? The main problem seems that there are local syncers that are mocked correctly, but a global syncer (`TrialRunner._syncer`) that tries to sync to AWS.

Reply:
this is fine for now